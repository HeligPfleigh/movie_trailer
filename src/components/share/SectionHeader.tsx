import React from 'react';
import {Box, Typography} from '../common';
import {StyleSheet, TouchableOpacity} from 'react-native';
import {colors, spacing} from '@movie_trailer/theme';
import AngleRightIcon from '@movie_trailer/assets/icons/AngleRight';

const styles = StyleSheet.create({
  divider: {
    height: '100%',
    width: 1,
    backgroundColor: colors.white,
    marginRight: spacing(0.5),
    marginLeft: spacing(1),
    opacity: 0.5,
  },
});

interface ISectionHeaderProps {
  title: string;
  subtitle?: string;
  onPress?: () => void;
}

const SectionHeader: React.FC<ISectionHeaderProps> = ({
  title,
  subtitle,
  onPress,
}: ISectionHeaderProps) => {
  return (
    <Box flex={false} row center mb={2}>
      <Typography variant="h5" fontFamily="Poppins-Bold" color={colors.white}>
        {title}
        {subtitle ? (
          <Typography variant="h5" color={colors.white}>
            {` ${subtitle}`}
          </Typography>
        ) : (
          ''
        )}
      </Typography>

      <Box flex={false} style={styles.divider} />

      <TouchableOpacity onPress={onPress}>
        <Box flex={false} row center>
          <Typography variant="caps1" color={colors.lavenderMagenta}>
            See All
          </Typography>
          <Box flex={false} ml={0.25}>
            <AngleRightIcon />
          </Box>
        </Box>
      </TouchableOpacity>
    </Box>
  );
};

export default SectionHeader;
