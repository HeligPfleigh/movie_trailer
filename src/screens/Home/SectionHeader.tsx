import React from 'react';
import {Box, Typography} from '@movie_trailer/components';
import {StyleSheet, TouchableOpacity} from 'react-native';
import {colors, spacing} from '@movie_trailer/theme';
import AngleRightIcon from '@movie_trailer/assets/icons/AngleRight';

const styles = StyleSheet.create({
  divider: {
    height: '100%',
    width: 1,
    backgroundColor: colors.white,
    marginRight: spacing(0.5),
    marginLeft: spacing(1),
    opacity: 0.5,
  },
});

const SectionHeader: React.FC = () => {
  return (
    <Box flex={false} row center>
      <Typography variant="h5" fontWeight="700" color={colors.white}>
        {'Popular '}
        <Typography variant="h5" fontWeight="400" color={colors.white}>
          Genres
        </Typography>
      </Typography>

      <Box flex={false} style={styles.divider} />

      <TouchableOpacity>
        <Box flex={false} row center>
          <Typography variant="caps1" color={colors.lavenderMagenta}>
            See All
          </Typography>
          <Box flex={false} ml={0.25}>
            <AngleRightIcon />
          </Box>
        </Box>
      </TouchableOpacity>
    </Box>
  );
};

export default SectionHeader;
