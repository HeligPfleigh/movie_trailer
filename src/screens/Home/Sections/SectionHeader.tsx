import React from 'react';
import {Box, Typography} from '@movie_trailer/components';
import {StyleSheet, TouchableOpacity} from 'react-native';
import {colors, spacing} from '@movie_trailer/theme';
import AngleRightIcon from '@movie_trailer/assets/icons/AngleRight';

const styles = StyleSheet.create({
  divider: {
    height: '100%',
    width: 1,
    backgroundColor: colors.white,
    marginRight: spacing(0.5),
    marginLeft: spacing(1),
    opacity: 0.5,
  },
});

interface ISectionHeaderProps {
  title: string;
  onPress?: () => void;
}

const SectionHeader: React.FC<ISectionHeaderProps> = ({
  title,
  onPress,
}: ISectionHeaderProps) => {
  const [emphasizeWord, ...others] = title.split(' ');

  return (
    <Box flex={false} row center mb={2}>
      <Typography variant="h5" fontWeight="700" color={colors.white}>
        {emphasizeWord}
        {others.length ? (
          <Typography variant="h5" fontWeight="400" color={colors.white}>
            {` ${others.join(' ')}`}
          </Typography>
        ) : (
          ''
        )}
      </Typography>

      <Box flex={false} style={styles.divider} />

      <TouchableOpacity onPress={onPress}>
        <Box flex={false} row center>
          <Typography variant="caps1" color={colors.lavenderMagenta}>
            See All
          </Typography>
          <Box flex={false} ml={0.25}>
            <AngleRightIcon />
          </Box>
        </Box>
      </TouchableOpacity>
    </Box>
  );
};

export default SectionHeader;
