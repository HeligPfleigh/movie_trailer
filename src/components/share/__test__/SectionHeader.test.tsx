import React from 'react';
import {fireEvent, render as rtlRender} from '@testing-library/react-native';

import SectionHeader from '../SectionHeader';

describe('SectionHeader', () => {
  test('should match snapshot', () => {
    const wrapper = rtlRender(<SectionHeader title="Popular" />);

    const seeAll = wrapper.queryByText(/see all/i);
    expect(seeAll).toBeTruthy();

    expect(wrapper).toMatchSnapshot();

    wrapper.rerender(<SectionHeader title="Popular" subtitle="Person" />);

    expect(wrapper).toMatchSnapshot();
  });

  test('see all button is clickable', () => {
    const onPress = jest.fn();
    const wrapper = rtlRender(
      <SectionHeader title="Popular" onPress={onPress} />,
    );

    const seeAll = wrapper.getByText(/see all/i);

    fireEvent.press(seeAll);
    expect(onPress).toHaveBeenCalled();
  });
});
