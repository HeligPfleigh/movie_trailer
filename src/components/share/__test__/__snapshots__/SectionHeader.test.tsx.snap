// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      false,
      Object {
        "flex": 0,
      },
      Object {
        "flexDirection": "row",
      },
      undefined,
      Object {
        "alignItems": "center",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      Object {},
      Object {},
      Object {
        "marginBottom": 16,
      },
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins",
        },
        Object {
          "color": "#ffffff",
        },
        undefined,
        undefined,
        Object {
          "fontFamily": "Poppins-Bold",
        },
        Object {
          "fontSize": 20,
          "lineHeight": 24,
        },
        undefined,
      ]
    }
  >
    Popular
    
  </Text>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        false,
        Object {
          "flex": 0,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "marginLeft": 8,
          "marginRight": 4,
          "opacity": 0.5,
          "width": 1,
        },
      ]
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          false,
          Object {
            "flex": 0,
          },
          Object {
            "flexDirection": "row",
          },
          undefined,
          Object {
            "alignItems": "center",
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins",
            },
            Object {
              "color": "#E884E4",
            },
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 15,
              "lineHeight": 26,
            },
            undefined,
          ]
        }
      >
        See All
      </Text>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            false,
            Object {
              "flex": 0,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {},
            Object {},
            Object {},
            Object {
              "marginLeft": 2,
            },
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            undefined,
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={16}
          bbWidth={16}
          fill="none"
          focusable={false}
          height={16}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          width={16}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M10.36 8.473 6.587 12.24a.666.666 0 0 1-1.14-.47c0-.176.069-.345.193-.47l3.3-3.333-3.3-3.3a.667.667 0 0 1 .217-1.088.667.667 0 0 1 .513 0 .667.667 0 0 1 .217.148l3.773 3.766a.666.666 0 0 1 0 .98Z"
              fill={4293428452}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;

exports[`SectionHeader should match snapshot 2`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      false,
      Object {
        "flex": 0,
      },
      Object {
        "flexDirection": "row",
      },
      undefined,
      Object {
        "alignItems": "center",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      Object {},
      Object {},
      Object {
        "marginBottom": 16,
      },
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      Object {},
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins",
        },
        Object {
          "color": "#ffffff",
        },
        undefined,
        undefined,
        Object {
          "fontFamily": "Poppins-Bold",
        },
        Object {
          "fontSize": 20,
          "lineHeight": 24,
        },
        undefined,
      ]
    }
  >
    Popular
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins",
          },
          Object {
            "color": "#ffffff",
          },
          undefined,
          undefined,
          undefined,
          Object {
            "fontSize": 20,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
    >
       Person
    </Text>
  </Text>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        false,
        Object {
          "flex": 0,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "marginLeft": 8,
          "marginRight": 4,
          "opacity": 0.5,
          "width": 1,
        },
      ]
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          false,
          Object {
            "flex": 0,
          },
          Object {
            "flexDirection": "row",
          },
          undefined,
          Object {
            "alignItems": "center",
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins",
            },
            Object {
              "color": "#E884E4",
            },
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 15,
              "lineHeight": 26,
            },
            undefined,
          ]
        }
      >
        See All
      </Text>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            false,
            Object {
              "flex": 0,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {},
            Object {},
            Object {},
            Object {
              "marginLeft": 2,
            },
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            undefined,
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={16}
          bbWidth={16}
          fill="none"
          focusable={false}
          height={16}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          width={16}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M10.36 8.473 6.587 12.24a.666.666 0 0 1-1.14-.47c0-.176.069-.345.193-.47l3.3-3.333-3.3-3.3a.667.667 0 0 1 .217-1.088.667.667 0 0 1 .513 0 .667.667 0 0 1 .217.148l3.773 3.766a.666.666 0 0 1 0 .98Z"
              fill={4293428452}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;
